src/Catty/ViewController/CatrobatTableViewController/SettingsTableViewController.swift

```diff
@@ -0,0 +1,211 @@
+            }))
+        }))
+    }
+
+    addSection(BOTableViewSection(headerTitle: "", handler: { section in
+        section?.addCell(BOChoiceTableViewCell(title: kLocalizedAboutPocketCode, key: "choice_2", handler: { cell in
+            if let aboutPocketCodeCellChoice = cell as? BOChoiceTableViewCell {
+                aboutPocketCodeCellChoice.destinationViewController = AboutPocketCodeOptionTableViewController()
+                aboutPocketCodeCellChoice.backgroundColor = UIColor.background
+                aboutPocketCodeCellChoice.mainColor = UIColor.globalTint
+            }
+        }))
+
+        section?.addCell(BOChoiceTableViewCell(title: kLocalizedTermsOfUse, key: "choice_2", handler: { cell in
+            if let termsOfUseCellChoice = cell as? BOChoiceTableViewCell {
+                termsOfUseCellChoice.destinationViewController = TermsOfUseOptionTableViewController()
+                termsOfUseCellChoice.backgroundColor = UIColor.background
+                termsOfUseCellChoice.mainColor = UIColor.globalTint
+            }
+        }))
+    }))
+
+    addSection(BOTableViewSection(headerTitle: "", handler: { section in
+        section?.addCell(BOButtonTableViewCell(title: kLocalizedPrivacySettings, key: nil, handler: { cell in
+                if let privacyCellButton = cell as? BOButtonTableViewCell {
+                    privacyCellButton.backgroundColor = UIColor.background
+                    privacyCellButton.mainColor = UIColor.globalTint
+                    privacyCellButton.actionBlock = {
+                        self.openPrivacySettings()
+                }
+            }
+        }))
+
+        section?.addCell(BOButtonTableViewCell(title: kLocalizedRateUs, key: nil, handler: { cell in
+                if let rateUsCellButton = cell as? BOButtonTableViewCell {
+                    rateUsCellButton.backgroundColor = UIColor.background
+                    rateUsCellButton.mainColor = UIColor.globalTint
+                    rateUsCellButton.actionBlock = {
+                        self.openRateUsURL()
+                }
+            }
+        }))
+
+        var version: String?
+        if let object = Bundle.main.infoDictionary?["CFBundleShortVersionString"] {
+            version = "\(kLocalizedVersionLabel)\(object) (\(String(describing: Util.appBuildVersion())))"
+        }
+
+    #if DEBUG
+        version = "\(version ?? "")(\(kLocalizedDebugMode))"
+    #endif
+
+        section?.footerTitle = version
+    }))
+}
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        UserDefaults.standard.synchronize()
+    }
+
+    @objc func changeFirebaseCrashReportSettings(_ sender: UISwitch?) {
+        NotificationCenter.default.post(name: NSNotification.Name(rawValue: NotificationName.settingsCrashReportingChanged), object: NSNumber(value: sender?.isOn ?? false))
+    }
+
+    func presentAlertController(withTitle title: String?, message: String?) {
```

Comment:
You can set all methods starting from here to _fileprivat_ since they will and should not be used from outside this file.